src/dispatch/incident/flows.py

```diff
@@ -865,8 +832,13 @@ def incident_closed_flow(incident_id: int, command: Optional[dict] = None, db_se
     )
 
     if incident.visibility == Visibility.open:
-        # we archive the artifacts in the storage
-        archive_incident_artifacts(incident, db_session)
+
+        # open file to domain on closure
+        if INCIDENT_STORAGE_OPEN_ON_CLOSE:
+            # we archive the artifacts in the storage
+            storage_plugin = plugins.get(INCIDENT_PLUGIN_STORAGE_SLUG)
+            storage_plugin.archive(folder_id=incident.storage.resource_id,)
+            log.debug("We have archived the incident artifacts.")
```

Comment:
I'd rather have this in a function like before.

Reply:
I don't think there is enough logic to justify a function.

Reply:
It would improve flow code readability. See functions `delete_participant_groups()` and `delete_conference()` below.